File: code-coaches/medium/secret-message/SecretMessage.cs
using System;

public class SecretMessage
{
    static string EncodeMessage(string message)
    {
        string textInvertedAscii = "";
        int asciiLowercaseA = 97;
        int asciiLowercaseZ = 122;

		foreach (char character in message)
		{
			if (char.IsLetter(character))
				textInvertedAscii += Convert.ToChar(asciiLowercaseZ - (int) char.ToLower(character) + asciiLowercaseA);
			else
				textInvertedAscii += character;
		}

		return textInvertedAscii;
    }

	static void Main(string[] args)
	{
		string message = Console.ReadLine();
		Console.WriteLine(EncodeMessage(message));
	}
}

File: code-coaches/medium/secret-message/SecretMessage.java
import java.util.Scanner;

public class SecretMessage {

    public static String encodeMessage(String message) {
        String textInvertedAscii = "";
        int asciiLowercaseA = 97;
        int asciiLowercaseZ = 122;

		for (char character : message.toCharArray()) {
			if (Character.isLetter(character)) {
				textInvertedAscii += Character.toString(asciiLowercaseZ - (int) Character.toLowerCase(character) + asciiLowercaseA);
			} else {
				textInvertedAscii += character;
			}
		}

		return textInvertedAscii;
    }

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		String message = input.nextLine();

		System.out.println(encodeMessage(message));

		input.close();
	}

}
